test/system/090-events.bats

```diff
@@ -129,3 +129,69 @@ EOF
     run cat $events_file
     is "$output" ".*\"Name\":\"$IMAGE" "test"
 }
+
+function _populate_events_file() {
+    # Create 100 duplicate entries to populate the events log file.
+    # The total file size is 9500 bytes.
+    for i in {0..99}; do echo '{"Name":"busybox","Status":"pull","Time":"2022-04-06T11:26:42.723667984+02:00","Type":"image","Attributes":null}' >> $1; done
+}
+
+@test "events log-file rotation" {
+   skip_if_remote "setting CONTAINERS_CONF logger options does not affect remote client"
+
+   # Make sure that the events log file is (not) rotated depending on the
+   # settings in containers.conf.
+
+    tmpdir=$PODMAN_TMPDIR/events-test
+    run mkdir -p $tmpdir
+
+    # Config without a limit
+    eventsFile=$tmpdir/events.txt
+    _populate_events_file $eventsFile
+    containersConf=$tmpdir/containers.conf
+    cat >$containersConf <<EOF
+[engine]
+events_logger="file"
+events_logfile_path="$eventsFile"
+EOF
+
+    # Create events *without* a limit and make sure that it has not been
+    # rotated/truncated.
+    contentBefore=$(head -n100 $eventsFile)
+    CONTAINERS_CONF=$containersConf run_podman run --rm $IMAGE true
+    contentAfter=$(head -n100 $eventsFile)
+    is "$contentBefore" "$contentAfter" "events file has not been rotated"
+
+    # Repopulate events file
+    rm $eventsFile
+    _populate_events_file $eventsFile
+
+    # Config with a limit
+    rm $containersConf
+    cat >$containersConf <<EOF
+[engine]
+events_logger="file"
+events_logfile_path="$eventsFile"
+# The limit of 4750 is the *exact* half of the inital events file.
+events_logfile_max_size=4750
+EOF
+
+    # Create events *with* a limit and make sure that it has been
+    # rotated/truncated.  Once rotated, the events file should only contain the
+    # second half of its previous events plus the new ones.
+    expetedContentAfterTruncation=$tmpdir/truncated.txt
+    head -n49 $eventsFile > $expetedContentAfterTruncation
+
+    contentBefore=$(head -n100 $eventsFile)
+    CONTAINERS_CONF=$containersConf run_podman pull $IMAGE
+    tail -n2 $eventsFile >> $expetedContentAfterTruncation
+
+    # Make sure the events file looks as expected.
+    is "$(cat $eventsFile)" "$(cat $expetedContentAfterTruncation)" "events file has been rotated"
+
+    # Make sure that `podman events` can read the file, and that it returns the
+    # same amount of events.  We checked the contents before.
+    CONTAINERS_CONF=$containersConf run_podman events --stream=false --since="2022-03-06T11:26:42.723667984+02:00"
+    is "$(wc -l $eventsFile)" "$(echo "$output" | wc -l) .*" "all events are returned"
```

Comment:
Try this instead:
```bash
    is "$(wc -l <$eventsFile)" "$(wc -l <<<$output)" "all events are returned"
```
Reasons:
* Any time you use `.*` is a time where things can go wrong. Precise comparisons are much better than hazy ones.
* Gets rid of nested double-quotes, which are hard to read